src/stackmap_checker/call_stack_state.h

```diff
@@ -30,21 +45,46 @@ typedef struct Frame {
 typedef struct CallStackState {
     frame_t *frames;
     uint32_t depth;
-    // The address of the return address of main.
-    uint64_t main_ret_addr;
-    // The BP of main.
-    uint64_t main_bp;
-    // The values of the general-purpose registers in main.
-    unw_word_t *main_regs;
 } call_stack_state_t;
 
+// The memory area which represents the restored call stack.
+typedef struct RestoredStackSegment {
```

Comment:
Does this represent a frame? I worry about the term "segment" as this has other meanings at the binary level.

Reply:
It represents the entire call stack, really. It's only used when inlining is detected. This is because the stack may need to be enlarged. To simplify things, it represents the entire 'reconstructed' stack, which is copied to the real stack before the recovery system returns.